org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/rename/RenameService.xtend

```diff
@@ -101,7 +101,9 @@ class RenameService implements IRenameService, IRenameServiceExtension {
 	protected def EObject getElementAtOffset(XtextResource xtextResource, Document document, Position caretPosition) {
 		val caretOffset = document.getOffSet(caretPosition)
 		val leafNode = NodeModelUtils.findLeafNodeAtOffset(xtextResource.parseResult.rootNode, caretOffset)
-		val offset = if (caretOffset > 0 && leafNode.offset === caretOffset && !isIdentifier(leafNode)) 
+		val offset = if (caretOffset > 0 &&
+			(caretOffset === xtextResource.parseResult.rootNode.endOffset 
```

Comment:
is this nullsafe?

Reply:
OK, I made it more defensive and replaced endOffset by totalEndOffset.